### PR TITLE
Add initial notes on how we (want to) do code review

### PR DESCRIPTION
**Pull request by tsibley**
Notes from our "How We Work" series of discussions.  Ideas here were
generally discussed with the whole team, I'm just putting them down into
words as I understand them.

This is a starting point to formalize practices we've adopted (or want
to adopt).  Corrections, clarifications, additions, removals, and other
changes are welcome.

### CODE REVIEW

**Review by kairstenfay — APPROVED** [review]
My code review to your meta-code review PR is... this looks fantastic, thank you! 
As someone who has gone through many rounds of your code review, I sincerely appreciate that you wrote this, @tsibley, so that I can reference it from time to time to make sure I'm employing best practices in my feedback.

**Comment by kairstenfay** [comment]
Oh! one more thing.. please don't forget to link to this page from the main README.

File: Code-review.md
Comment (kschwabefry) [inline Code-review.md:22]: Had to look up what a punch list was 😄

**Review by kschwabefry — APPROVED** [review]
Looks great, thanks for drafting this up Tom!

**Review by joverlee521 — APPROVED** [review]
Thank you for writing this up Tom!

I want to add that we've sometimes done in-person pair code reviews during pre-pandemic times. It would be great to explicitly state that as an option. Like you said, text-only communication can be hard. A quick meeting/video chat can reduce a lot of back and forth in code review.

File: Code-review.md
Comment (jccraft-uw) [inline Code-review.md:53]: Is there an easy way we can do this? 
In the "Review changes" workflow in github the option is listed as
"Request changes
Submit feedback that must be addressed before merging."
When I see requests in there I read them as required before merging.

For suggestions that are not merge blockers would it make sense to add the comments into the comments box at the center bottom of the page when you're on the Conversation view?

File: Code-review.md
Comment (tsibley) [reply Code-review.md:22]: Ahhh, would it be better to use "check list" instead?

Comment (tsibley) [reply Code-review.md:22]: Alternatively, could keep it but link to https://m-w.com/dictionary/punch+list?

File: Code-review.md
Comment (tsibley) [reply Code-review.md:55]: GitHub offers three statuses for the overall state of the review (i.e. not tied to specific review comments):

- Commenting only
- Requesting changes
- Approving as-is

A review may have comments that are both blockers and not blockers, but the overall review status can't indicate which those are.

My tendency would be to leave this to the flexibility of written words instead of trying to codify it into a process.

**Comment by tsibley** [comment]
@kairstenfay Linked from _Home.md_ now, thanks for the reminder.

@joverlee521 Good point, added a sentence about video calls. :+1:

File: Code-review.md
Comment (kschwabefry) [reply Code-review.md]: I think the specificty is nice. although checklist is more commonly used. I'm happy with a link to MW to preempt confusion :)

**Pull request merged**
